includes/class-newspack-newsletters.php

```diff
@@ -856,6 +875,15 @@ public static function retrieve_data( $id ) {
 				}
 			}
 
+			$tags = [];
+			if ( $list_id ) {
+				$tags_response = self::validate_mailchimp_operation(
+					$mc->get( "lists/$list_id/segments" ),
```

Comment:
According to the [Mailchimp docs](https://mailchimp.com/developer/reference/lists/list-segments/) this will only return up to 10 tags. I don't really have an understanding of how tags are typically used so its hard to guess how many there might be, but it might make sense to make the count pretty high just to be safe.

Reply:
Good catch, fixed in df1cc478af262f1dfd3cace2d2297f3479a6a2ab